docs/conventions/component.md

```diff
@@ -0,0 +1,461 @@
+This main mixin should be named following the component's main class.
+
+Example:
+```scss
+@mixin ecl-navigation-menu-common() {
+  // Some css rules or mixin include
+}
+@mixin ecl-navigation-menu-responsive() {
+  // Some css rules
+}
+@mixin ecl-navigation-menu() {
+  @include ecl-navigation-menu-common();
+  @include ecl-navigation-menu-responsive();
+}
+```
+
+The css rules in generic components should not define specific display elements (like color, background, and more globally everything specific to a system).
+
+If display is different in several systems, these differences should be put as parameters in the mixin. The goal here is to be able to call the same generic code, and just pass some parameters to alter display. There is no need to repeat component's name for these parameters, as they are limited to this mixin.
+
+Example:
+```scss
+@mixin ecl-navigation-menu(
+  $bar-bg-mobile,
+  $bar-bg-desktop,
+  $toggle-btn-color,
+  $group-bg,
+  $separator-mobile,
+  $link-color,
+  $active-color,
+  $active-bg
+) {
+  // Some css rules or mixin include
+}
+```
+
+### Systems component
+
+SCSS files in system-specific components rely on generic SCSS file.
+It contains 4 main parts
+
+#### Import base and generic mixins
+
+The file should import generic rules located in `@ecl/{system}-base`, and the ones of the generic component.
+
+#### Check dependencies calls
+
+If the component override some other components (like links, buttons, ...), it should ensure that these dependencies are loaded ahead.
+This is done by using the mixin `check-imports`.
+
+#### Generic mixin use
+
+Call of the mixin defined in generic component, with custom parameters.
+
+#### Additional css rules
+
+If needed, some extra css rules could be added after to specialize the component.
+
+Example:
+```scss
+// Import base and generic
+@import '@ecl/ec-base';
+@import '@ecl/generic-component-navigation-menu/generic-component-navigation-menu';
+
+// Check if overridden dependencies are already loaded, if needed
+@include check-imports(('ec-component-link', 'ec-component-button'));
+
+// Use generic mixin
+@include exports('ec-component-navigation-menu') {
+  @include ecl-navigation-menu(
+    $bar-bg-mobile: map-get($ecl-colors, 'grey-15'),
+    $bar-bg-desktop: map-get($ecl-colors, 'grey-10'),
+    $toggle-btn-color: map-get($ecl-colors, 'blue-100'),
+    $group-bg: map-get($ecl-colors, 'grey-5'),
+    $separator-mobile: map-get($ecl-colors, 'grey-15'),
+    $link-color: map-get($ecl-colors, 'blue-100'),
+    $active-color: map-get($ecl-colors, 'blue-100'),
+    $active-bg: map-get($ecl-colors, 'grey-10')
+  );
+
+  // Add custom rules
+  .ecl-navigation-menu__item--active {
+    [...]
+  }
+  .ecl-navigation-menu__group::before {
+    [...]
+  }
+}
+```
+
+### Specific case: editor preset
+
+If the component alters a default html tag display (like links, blockquotes, ...), it should also provide css rules for editing.
```

Comment:
`default html tag display` is a bit vague, I think you mean HTML elements not having CSS classes with ECL namespace.